src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt

```diff
@@ -2228,10 +2228,17 @@ SelectExpressionItem SelectExpressionItem():
     Alias alias = null;
 }
 {
-      expression=Expression() { selectExpressionItem = new SelectExpressionItem(); selectExpressionItem.setExpression(expression); }
+      (
+        LOOKAHEAD( Condition() ) expression = Condition()
```

Comment:
Does this indeed improve parsing speed? Lookahead a ´Conditon´ should be quite expensive as well. I am just curious.

Reply:
Greetings.

I don't think it will improve parsing speed (and I did never mention anything like that).
But it fixes Correctness for stuff like `CASES WHEN ... THEN ... END + 1`.

Honestly, I feel like it is just by-passing the problem and something is wrong with the predating order of the Expressions. However, I spent hours on that and have not been able to understand what/why exactly goes wrong.

This Grammar needs a serious rewrite, which will never happen because it still works "just good enough" and it was a mammoth task.

Reply:
Right you are.